### DISCUSSION TITLE
How to prevent creating link vertices on double clicking link labels?

### DISCUSSION THREAD

**Discussion by fafeitsch** [clientIO/joint#2370] [Q&A] [answered]
### Introduction

We are using Joint to create the following graph: 

![image](https://github.com/clientIO/joint/assets/842642/5a8f693e-e66e-4fcd-927e-d680b3acd175)

We have registered a double click handler on the link label "Yes" and I have selected the link in the screen shot.

Normally, if double click on the link label, an inline editor opens and I can edit the link. However, in the depicted situation, if I happen to double click in the lower right corner of the link label, a vertex is created on the link before the inline editor is opened:

![image](https://github.com/clientIO/joint/assets/842642/9d2e3f39-771e-4540-9c1c-914396a8a6b5)

This happens because the link tools are drawn invisibly on top of the whole graph, thus creating the single click on every double click event on the graph.

How can I avoid that double clicking on a link label which covers the actual link accidentally creates vertices? 

### Restrictions & Constraints

1. We can't disable the vertex feature (omiting `new joint.linkTools.Vertices()`), because we like the vertex feature in general.
2. Using `snapRadius` doesn't solve the problem.
3. Using `layer: null` on `ToolsView`-creation and adding `setZ(1000)` to all link cells doesn't solve the problem either. The Tools-Layer is now placed inside the graph layer, but still after (and, thus, on top) of the graph and capturing the mouse events
4. We have already tried to stop click propagation in various places but didn't succeed yet.

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS

**Comment by kumilingus** [comment, marked as answer]
The only option now I can think of is to set the `labelsLayer` paper's option to `true` and move the label's layer `<g/>` element after the tool's layer element.

```js
const paper = new dia.Paper({
   /* ... */
   labelsLayer: true
});
// Move the labels layer in front of the tools layer
// so the labels are not covered by the tools.
const labelsLayerEl = paper.getLayerNode(dia.Paper.Layers.LABELS);
labelsLayerEl.parentNode.appendChild(labelsLayerEl);
```

*Note: This will be more elegant once we introduce an official Layer API.*

**Reply by fafeitsch** [reply]
Nice, seems to work. However, `dia.paper.Layers.LABELS` didn't work for us, no idea why. Using the plain string `'labels'` did work, though.

Thank you!
